### PR TITLE
Reporting multirange datatypes in analyze schema and assessment report

### PR DESCRIPTION
**Pull request by ShivanshGahlot**
### Describe the changes in this pull request
Added support to report multirange datatypes in Postgres in assessment report and analyze-schema:
```
int4multirange	Multiple integer ranges
int8multirange	Multiple bigint ranges
nummultirange	Multiple numeric ranges
tsmultirange	Multiple timestamp ranges
tstzmultirange	Multiple timestamptz ranges
datemultirange	Multiple date ranges
```
Modified tests to validate the same.

Issue:
https://yugabyte.atlassian.net/browse/DB-14218?atlOrigin=eyJpIjoiMTBiMGM2ZGQ4MTEzNDBmMTk3MTMwYWMwZGQ1NjIyMzAiLCJwIjoiaiJ9

### Describe if there are any user-facing changes

Assessment Report will include these whenever needed:
image

Analyze schema report will include these whenever needed:
image

### How was this pull request tested?

Tested it locally. 
Added unit tests to: `issues_ddl_test.go`
Added cases in the end to end tests for analyze schema and migration assessment

### Does your PR have changes that can cause upgrade issues? 
| Component        | Breaking changes? |
| :----------------------------------------------: | :-----------: |
| MetaDB                                                    |  No |
| Name registry json                                        |  No |
| Data File Descriptor Json                                 | No |
| Export Snapshot Status Json                               |  No |
| Import Data State                                         | No |
| Export Status Json                                        |  No |
| Data .sql files of tables                                 |  No |
| Export and import data queue                              |  No |
| Schema Dump                                               |  No |
| AssessmentDB                                              | No |
| Sizing DB                                                 |  No |
| Migration Assessment Report Json                          |  No |
| Callhome Json                                             |  No |
| YugabyteD Tables                                          |  No |
| TargetDB Metadata Tables                                  |  No |

### CODE REVIEW

File: yb-voyager/src/query/queryissue/issues_ddl.go [hunk 1]
```diff
@@ -438,4 +438,18 @@ func NewLODatatypeIssue(objectType string, objectName string, SqlStatement strin
 	issue := loDatatypeIssue
 	issue.TypeName = fmt.Sprintf("%s on column - %s", issue.TypeName, colName)
 	return newQueryIssue(issue, objectType, objectName, SqlStatement, map[string]interface{}{})
-}
\ No newline at end of file
+}
+
+var multiRangeDatatypeIssue = issue.Issue{
+	Type:       MULTI_RANGE_DATATYPE,
+	TypeName:   "Unsupported datatype - multirange",
```
Comment (priyanshi-yb) [inline yb-voyager/src/query/queryissue/issues_ddl.go:445]: "Unsupported datatype"

File: yb-voyager/src/query/queryissue/issues_ddl.go [hunk 2]
```diff
@@ -438,4 +438,18 @@ func NewLODatatypeIssue(objectType string, objectName string, SqlStatement strin
...
+	Suggestion: "Multirange data type is not yet supported in YugabyteDB, no workaround available currently",
+	GH:         "", //TODO
+	DocsLink:   "", //TODO
+}
+
+func NewMultiRangeDatatypeIssue(objectType string, objectName string, sqlStatement string, typeName string, colName string) QueryIssue {
+	issue := multiRangeDatatypeIssue
+	issue.TypeName = fmt.Sprintf("%s on column - %s", issue.TypeName, colName)
```
Comment (priyanshi-yb) [inline yb-voyager/src/query/queryissue/issues_ddl.go:453]: Add the respective type
issue.TypeName = fmt.Sprintf("%s - %s on column - %s", issue.TypeName, typeName, colName)

File: yb-voyager/src/srcdb/postgres.go [hunk 3]
```diff
@@ -50,7 +50,7 @@ const PG_STAT_STATEMENTS = "pg_stat_statements"
 
 var pg_catalog_tables_required = []string{"regclass", "pg_class", "pg_inherits", "setval", "pg_index", "pg_relation_size", "pg_namespace", "pg_tables", "pg_sequences", "pg_roles", "pg_database", "pg_extension"}
 var information_schema_tables_required = []string{"schemata", "tables", "columns", "key_column_usage", "sequences"}
-var PostgresUnsupportedDataTypes = []string{"GEOMETRY", "GEOGRAPHY", "BOX2D", "BOX3D", "TOPOGEOMETRY", "RASTER", "PG_LSN", "TXID_SNAPSHOT", "XML", "XID", "LO"}
+var PostgresUnsupportedDataTypes = []string{"GEOMETRY", "GEOGRAPHY", "BOX2D", "BOX3D", "TOPOGEOMETRY", "RASTER", "PG_LSN", "TXID_SNAPSHOT", "XML", "XID", "LO", "INT4MULTIRANGE", "INT8MULTIRANGE", "NUMMULTIRANGE", "TSMULTIRANGE", "TSTZMULTIRANGE", "DATEMULTIRANGE"}
```
Comment (priyanshi-yb) [inline yb-voyager/src/srcdb/postgres.go:53]: Add these to the `PostgresUnsupportedDataTypesForDbzm` as well.

File: migtests/tests/analyze-schema/dummy-export-dir/schema/tables/table.sql [hunk 4]
```diff
@@ -384,4 +384,82 @@ CREATE TABLE public.locations (
...
+ALTER TABLE ONLY numeric_multirange_table
+    ADD CONSTRAINT numeric_multirange_table_pkey PRIMARY KEY (id);
+
+
+ALTER TABLE ONLY timestamp_multirange_table
+    ADD CONSTRAINT timestamp_multirange_table_pkey PRIMARY KEY (id);
+
+
+ALTER TABLE ONLY timestamptz_multirange_table
+    ADD CONSTRAINT timestamptz_multirange_table_pkey PRIMARY KEY (id);
```
Comment (sanyamsinghal) [inline migtests/tests/analyze-schema/dummy-export-dir/schema/tables/table.sql:465]: lets merge this into CREATE TABLE ddl unless required to be like this.

File: migtests/tests/analyze-schema/dummy-export-dir/schema/tables/table.sql [hunk 5]
```diff
@@ -384,4 +384,82 @@ CREATE TABLE public.locations (
...
+
+
+CREATE TABLE timestamptz_multirange_table (
+    id integer NOT NULL,
+    global_event_times tstzmultirange
+);
+
+
+ALTER TABLE ONLY bigint_multirange_table ALTER COLUMN id SET DEFAULT nextval('multirange_demo.bigint_multirange_table_id_seq'::regclass);
+
```
Comment (sanyamsinghal) [inline migtests/tests/analyze-schema/dummy-export-dir/schema/tables/table.sql:427]: same comment as above.

Comment (sanyamsinghal) [reply migtests/tests/analyze-schema/dummy-export-dir/schema/tables/table.sql:427]: actually you can even get rid of it, looks like we are not making use of these sequence/defaultvals

File: yb-voyager/src/query/queryissue/issues_ddl.go (hunk 1 above)
Comment (ShivanshGahlot) [reply yb-voyager/src/query/queryissue/issues_ddl.go]: Done

File: yb-voyager/src/query/queryissue/issues_ddl.go (hunk 2 above)
Comment (ShivanshGahlot) [reply yb-voyager/src/query/queryissue/issues_ddl.go]: Done

File: yb-voyager/src/srcdb/postgres.go (hunk 3 above)
Comment (ShivanshGahlot) [reply yb-voyager/src/srcdb/postgres.go:53]: Done

File: migtests/tests/analyze-schema/dummy-export-dir/schema/tables/table.sql (hunk 5 above)
Comment (ShivanshGahlot) [reply migtests/tests/analyze-schema/dummy-export-dir/schema/tables/table.sql:427]: Yeah removed the sequence/defaultvals and merged Primary key into the create tables itself.

File: migtests/tests/analyze-schema/dummy-export-dir/schema/tables/table.sql (hunk 4 above)
Comment (ShivanshGahlot) [reply migtests/tests/analyze-schema/dummy-export-dir/schema/tables/table.sql:465]: Done

Comment (priyanshi-yb) [reply migtests/tests/analyze-schema/dummy-export-dir/schema/tables/table.sql:465]: File doesn't seem to be changed for these DDLs @ShivanshGahlot ?

**Review by priyanshi-yb — APPROVED** [review]
LGTM!

**Pull request merged**